contribute/design/1.x/cstor_cr_schema/apis.md

```diff
@@ -0,0 +1,550 @@
+type PoolSpec struct {
+	// NodeSelector is the labels that will be used to select
+	// a node for pool provisioning.
+	// Required field
+	NodeSelector map[string]string `json:"nodeSelector"`
+	// RaidConfig is the raid group configuration for the given pool.
+	RaidGroups []RaidGroup `json:"raidGroups"`
+	// PoolConfig is the default pool config that applies to the
+	// pool on node.
+	PoolConfig PoolConfig `json:"poolConfig"`
+}
+
+// PoolConfig is the default pool config that applies to the
+// pool on node.
+type PoolConfig struct {
+	// Cachefile is used for faster pool imports
+	// optional -- if not specified or left empty cache file is not
+	// used.
+	CacheFile string `json:"cacheFile"`
+	// DefaultRaidGroupType is the default raid type which applies
+	// to all the pools if raid type is not specified there
+	// Compulsory field if any raidGroup is not given Type
+	DefaultRaidGroupType string `json:"defaultRaidGroupType"`
+
+	// OverProvisioning to enable over provisioning
+	// Optional -- defaults to false
+	OverProvisioning bool `json:"overProvisioning"`
+	// Compression to enable compression
+	// Optional -- defaults to off
+	// Possible values : lz, off
+	Compression string `json:"compression"`
+	// Resources are the compute resources required by the cstor-pool
+	// container.
+	Resources *corev1.ResourceRequirements `json:"resources"`
+	// AuxResources are the compute resources required by the cstor-pool pod
+	// side car containers.
+	AuxResources *corev1.ResourceRequirements `json:"auxResources"`
+	// Tolerations, if specified, the pool pod's tolerations.
+	Tolerations []corev1.Toleration `json:"tolerations"`
+
+	// PriorityClassName if specified applies to this pool pod
+	// If left empty, DefaultPriorityClassName is applied.
+	// (See CStorPoolClusterSpec.DefaultPriorityClassName)
+	// If both are empty, not priority class is applied.
+	PriorityClassName string `json:"priorityClassName"`
+}
+
+// RaidGroup contains the details of a raid group for the pool
+type RaidGroup struct {
+	// Type is the raid group type
+	// Supported values are : stripe, mirror, raidz and raidz2
+
+	// stripe -- stripe is a raid group which divides data into blocks and
+	// spreads the data blocks across multiple block devices.
+
+	// mirror -- mirror is a raid group which does redundancy
+	// across multiple block devices.
+
+	// raidz -- RAID-Z is a data/parity distribution scheme like RAID-5, but uses dynamic stripe width.
+	// radiz2 -- TODO
+	// Optional -- defaults to `defaultRaidGroupType` present in `PoolConfig`
+	Type string `json:"type"`
+
+	// LoadType can have following values :
+	// readCache -- The raid group is a read cache.
+	// writeCache -- The raid group is a write cache.
+	// spare -- The raid group is spare.
+	// "" -- Empty value means this will be used a regular vdev to form pool.
+	// What about default value ? Can this be changed after some point of time.
+	LoadType string `json:"load_type"`
```

Comment:
better name is required

Reply:
Can we have something like `DeviceType` or `DeviceDesignation` which will tell us was that device is used for like readCache, writeCache, spare or normal device?

Reply:
Thanks